src/test/javascript/spec/service/orion/orion-connector.service.spec.ts

```diff
@@ -0,0 +1,240 @@
+    });
+
+    it('should forward onBuildStarted', () => {
+        serviceUnderTest.onBuildStarted('problem');
+
+        expect((window as any).orionBuildConnector.onBuildStarted).toHaveBeenCalledTimes(1);
+        expect((window as any).orionBuildConnector.onBuildStarted).toHaveBeenCalledWith('problem');
+    });
+
+    it('should forward onBuildFinished', () => {
+        serviceUnderTest.onBuildFinished();
+
+        expect((window as any).orionBuildConnector.onBuildFinished).toHaveBeenCalledTimes(1);
+        expect((window as any).orionBuildConnector.onBuildFinished).toHaveBeenCalledWith();
+    });
+
+    it('should forward onBuildFailed', () => {
+        serviceUnderTest.onBuildFailed([{ fileName: 'file', row: 5, column: 4, text: 'error', type: 'error', timestamp: 0 } as Annotation]);
+
+        expect((window as any).orionBuildConnector.onBuildFailed).toHaveBeenCalledTimes(1);
+        expect((window as any).orionBuildConnector.onBuildFailed).toHaveBeenCalledWith(
+            '{"errors":{"file":[{"row":5,"column":4,"text":"error","type":"error","ts":0}]},"timestamp":0}',
+        );
+    });
+
+    it('should forward onBuildFinished', () => {
+        serviceUnderTest.onTestResult(true, 'name', 'message');
+
+        expect((window as any).orionBuildConnector.onTestResult).toHaveBeenCalledTimes(1);
+        expect((window as any).orionBuildConnector.onTestResult).toHaveBeenCalledWith(true, 'name', 'message');
+    });
+
+    it('should forward isBuilding', () => {
+        let localState: OrionState = {} as any;
+        serviceUnderTest.state().subscribe((state) => {
+            localState = state;
+        });
+
+        serviceUnderTest.isBuilding(true);
+
+        expect(localState).toContainEntry(['building', true]);
+    });
+
+    it('should forward isCloning', () => {
+        let localState: OrionState = {} as any;
+        serviceUnderTest.state().subscribe((state) => {
+            localState = state;
+        });
+
+        serviceUnderTest.isCloning(true);
+
+        expect(localState).toContainEntry(['cloning', true]);
+    });
+
+    it('should navigate on startedBuildInOrion', () => {
+        const navigateSpy = jest.spyOn(router, 'navigateByUrl');
+
+        serviceUnderTest.startedBuildInOrion(5, 10);
+
+        expect(navigateSpy).toHaveBeenCalledTimes(1);
+        expect(navigateSpy).toHaveBeenCalledWith('/courses/5/exercises/10?withIdeSubmit=true');
+    });
+
+    it('should throw error in updateAssessment if no component present', () => {
+        const errorSpy = jest.spyOn(alertService, 'error');
+
+        serviceUnderTest.updateAssessment(5, '');
+
+        expect(errorSpy).toHaveBeenCalledTimes(1);
+        expect(errorSpy).toHaveBeenCalledWith('artemisApp.orion.assessment.updateFailed');
+    });
+
+    it('should forward in updateAssessment if component present', () => {
+        serviceUnderTest.activeAssessmentComponent = { updateFeedback: jest.fn() } as any;
```

Comment:
Shouldn't such component be mocked in the testbed configuration?

Reply:
Not really; This is a bit of a cheat where the activeAssessmentComponent is setting itself in the service to receive calls from the plugin (otherwise some weird workaround would be needed so the service can find whatever component is currently displayed to the user). But from the view of the service, this variable is normally undefined, only in some cases its set. So I think only setting it when testing these cases makes sense

Reply:
Is this documented at some place? I think it would make sense to reference it then.

Reply:
I added another comment in the service, if someone has difficulties understanding the test they should be able to find it immediately